src/app/shared/ui-components/error-message/error-message.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { ValidationService } from 'app/services';
+
+
+@Component({
+  selector: 'bw-error-message',
+  template: `<div class="alert alert-danger" *ngIf="errorMessage !== null">{{errorMessage}}</div>`
+})
+export class ErrorMessageComponent {
+  @Input() control: FormControl;
+  constructor() { }
+
+  get errorMessage() {
```

Comment:
why did you use `get` here ?

Reply:
to check if the user touched the form or any action happens to show error messages automatically without creating a listener